keps/sig-storage/1979-object-storage-support/README.md

```diff
@@ -396,43 +439,66 @@ NOTE: the contents of the files served in mountPath will be a COSI generated fil
 
-###### 2. The secret containing BucketInfo is mounted in the specified directory
+2.  **The secret containing BucketInfo is mounted in the specified directory**
 
 The above volume definition will prompt kubernetes to retrieve the secret and place it in the volumeMount path defined above. The contents of the secret will be of the format shown below:
 
+###### Single protocol example with keys
+
+```yaml
+apiVersion: v1
+kind: Secret
+metadata:
+  name: <BucketAccess.spec.credentialsSecretName>
+  namespace: <BucketAccess.metadata.namespace>
+data:
+  COSI_BUCKET_NAME: bc-$uuid
+  COSI_AUTHENTICATION_TYPE: KEY
+  COSI_PROTOCOLS: s3
+  COSI_S3_ENDPOINT: https://s3.amazonaws.com
+  COSI_S3_ACCESS_KEY_ID: AKIAIOSFODNN7EXAMPLE
+  COSI_S3_ACCESS_SECRET_KEY: wJalrXUtnFEMI/K...
+  COSI_S3_REGION: us-west-1
 ```
-    bucket_info.json
-    |-----------------------------------------------|
-    | {                                             |
-    |   apiVersion: "v1alpha1",                     |
-    |   kind: "BucketInfo",                         | 
-    |   metadata: {                                 |
-    |       name: "bc-$uuid"                        |
-    |   },                                          |
-    |   spec: {                                     |
-    |       bucketName: "bc-$uuid",                 |
-    |       authenticationType: "KEY",              |
-    |       endpoint: "https://s3.amazonaws.com",   |
-    |       accessKeyID: "AKIAIOSFODNN7EXAMPLE",    |
-    |       accessSecretKey: "wJalrXUtnFEMI/K...",  |
-    |       region: "us-west-1",                    |
-    |       protocols: [                            |
-    |         "s3"                                  |
-    |       ]                                       |
-    |   }                                           |
-    | }                                             |
-    |-----------------------------------------------|
 
+###### Multi-protocol example with keys
+
+<!-- TODO: what if one protocol supports IAM-style but not the other?
+           do we need auth type individually per-protocol?
+           do we advertize KEY if any don't support IAM? -->
+<!-- TODO: is IAM an object-storage-wide term, or is it S3-specific?
+           Can we agree on a generic terminology, like type=ServiceAccount? -->
+```yaml
+apiVersion: v1
+kind: Secret
+metadata:
+  name: <BucketAccess.spec.credentialsSecretName>
+  namespace: <BucketAccess.metadata.namespace>
+data:
+  COSI_BUCKET_NAME: bc-$uuid
+  COSI_AUTHENTICATION_TYPE: KEY
+  COSI_PROTOCOLS: Azure,S3
+  COSI_AZURE_ENDPOINT: https://blob.mysite.com
+  COSI_AZURE_ACCESS_TOKEN: AZURETOKENEXAMPLE
+  COSI_AZURE_EXPIRY_TIMESTAMP: 2006-01-02T15:04:05Z07:00
+  COSI_S3_ENDPOINT: https://s3.mysite.com
+  COSI_S3_ACCESS_KEY_ID: AKIAIOSFODNN7EXAMPLE
+  COSI_S3_ACCESS_SECRET_KEY: wJalrXUtnFEMI/K...
+  COSI_S3_REGION: us-west-1
```

Comment:
@shanduur here's the more complex version of what new secret might look like. Lmk if you have feedback.

Reply:
I like it! Little bit more date, and still has clear naming convention.

One more thing we should consider is Google Cloud Storage support. It is a valid value for the `Protocols`, yet there is no place for adding any credentials to secret for GCS.

1. Do we want to continue the support GCS in the COSI spec?
1. Do we need to consider additional fields for GCS, or do we leave it as?